python_modules/airline-demo/airline_demo/types.py

```diff
@@ -20,8 +24,54 @@
 )
 
 
+class SparkDataFrameSerializationStrategy(SerializationStrategy):
+    def serialize_value(self, _context, value, write_file_obj):
+        parquet_dir = tempfile.mkdtemp()
+        shutil.rmtree(parquet_dir)
+        value.write.parquet(parquet_dir)
+
+        archive_file_obj = tempfile.NamedTemporaryFile(delete=False)
+        try:
+            archive_file_obj.close()
+            archive_file_path = archive_file_obj.name
+            zipfile_path = shutil.make_archive(archive_file_path, 'zip', parquet_dir)
+            try:
+                with open(zipfile_path, 'rb') as archive_file_read_obj:
+                    write_file_obj.write(archive_file_read_obj.read())
+            finally:
+                try:
```

Comment:
I know this is more "pythonic" but can we just do an if check for existence instead?